src/source.cc

```diff
@@ -79,11 +100,35 @@ std::set<cyclus::BidPortfolio<cyclus::Material>::Ptr> Source::GetMatlBids(
     Request<Material>* req = *it;
     Material::Ptr target = req->target();
     double qty = std::min(target->quantity(), max_qty);
-    Material::Ptr m = Material::CreateUntracked(qty, target->comp());
-    if (!outrecipe.empty()) {
-      m = Material::CreateUntracked(qty, context()->GetRecipe(outrecipe));
+
+    // calculate packaging
+    double bid_qty = context()->GetPackage(package)->GetFillMass(qty);
+    int n_full_bids = static_cast<int>(std::floor(qty / bid_qty));
+    Package::ExceedsSplitLimits(n_full_bids);
+
+    std::vector<double> bids;
+    bids.assign(n_full_bids, bid_qty);
+
+    double remaining_qty = qty - (n_full_bids * bid_qty);
+    if ((remaining_qty > cyclus::eps()) && (remaining_qty >= context()->GetPackage(package)->fill_min())) {
+      bids.push_back(remaining_qty);
+    }
```

Comment:
If you created an untracked material of size `qty` and composition matching the request, couldn't you just use `Material::Package<Material>()` here?

Reply:
We're not packaging here because there's no guarantee the trade will go through, or that the trade will be of this exact size

Reply:
I see... but you are *effectively* doing the same calc that is done in `Resource::Package<>`, but in a different way.  I just wonder if there is a way for this to be done once and reused.

In `Resource::Package<>` you don't explicitly create a vector of individual package quantities like you do here.  Instead, you extract resources from the original material at this quantity until you are done.

This code here, is similar to (probably inspired by...) the code in `matl_sell_policy`.  It seems like this is an opportunity for streamlining.  I do see some use-case specific logic that creeps in.  I'll make an issue.

Reply:
Yes, I think the better long-term solution is to move archetypes to the sell policy when possible rather than repeating the same code, as I basically did here. If I was a few years earlier in my PhD when I realized that only Storage used the buy/sell policies, I probably would've targeted moving as many of the archetypes as possible onto those policies. I do think that would be an improvement to cycamore as a whole

Reply:
And while you don't package here, you do make a bunch of new untracked material objects from this below

Reply:
I made an issue in Cyclus to review this another time.